tokio/src/sync/batch_semaphore.rs

```diff
@@ -271,6 +271,7 @@ impl Semaphore {
                     Self::MAX_PERMITS
                 );
                 let prev = self.permits.fetch_add(rem << Self::PERMIT_SHIFT, Release);
+                let prev = prev >> Self::PERMIT_SHIFT;
```

Comment:
This fix should probably be mentioned in the 1.5 changelog. I discovered this bug the hard way :smile: 
cc @Darksonn

Reply:
Thanks. Can you open a PR to modify the changelog?

Reply:
Opened here https://github.com/tokio-rs/tokio/pull/3735